detox/src/devices/runtime/drivers/android/AndroidDriver.js

```diff
@@ -394,6 +407,30 @@ class AndroidDriver extends DeviceDriverBase {
       'Press any key to continue...'
     );
   }
+
+  async _clearAppData(bundleId) {
+    await this.adb.clearAppData(this.adbName, bundleId);
+  }
+
+  async _isAlreadyInstalled(bundleId, hash) {
+    return hash ? this.hashHelper.isRemoteHashEqualToLocal(this.adbName, bundleId, hash) : false;
```

Comment:
Technically, this logic is not enough: `hash_exists_and_equal <=> app installed`. For example, a user can manually uninstall the app, and still this would return `true`.
What's missing here, I suppose, is the preliminary check of whether the package is in fact installed on the device (a la `adb.isPackageInstalled()`...

Reply:
@jonathanmos still applicable